### PR TITLE
modifications to support scala 2.12

### PR DESCRIPTION
**Pull request by axos88**

### CODE REVIEW

File: src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ class AngularExecutionContext(
       case t: Throwable => reportFailure(t)
     }
 
-    timeout(run _, 0)
```
Comment (axos88) [inline src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala]: This should work, but borks on scala 2.12 unfortunately. probably a compiler bug. trying to repro and report.

**Comment by axos88** [comment]
#104

File: src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala (hunk 1 above)
Comment (mysticfall) [inline src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala]: Probably, we can just use `() => run()` instead? Or maybe we can specify the return type of the `run` method, by changing it to `def run(): Unit = try {...` or `def run() {...`. Please let me know how you think. Thanks!

Comment (axos88) [inline src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala]: () => run() works. should I update the PR to include that?

Comment (mysticfall) [inline src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala]: Please do :)

Comment (axos88) [inline src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala]: related: https://issues.scala-lang.org/browse/SI-7187

Comment (axos88) [inline src/main/scala/com/greencatsoft/angularjs/AngularExecutionContext.scala]: updated

**Pull request merged**